### PR TITLE
 Fixed issue [security] #19134: User with "Users" permission only can add arbitrary users to the not owned user group

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: protect the action in same way than UserGroupController
 Dev: filter usergroups dropdown in same way

### CODE REVIEW

**Comment by Shnoulle** [comment]
Minimal fix : all of this need more detail …

Here : for example : UserGroup/create : no way to create group

**Review by gabrieljenik — APPROVED** [review]
Code looks OK

File: application/controllers/UserManagementController.php [hunk 1]
```diff
@@ -1114,18 +1114,29 @@ public function actionBatchAddGroup()
...
-                $aResults[$sItem]['title'] = $model->users_name;
-                if (!$oUserGroup->hasUser($sItem)) {
-                    $aResults[$sItem]['result'] = $oUserGroup->addUser($sItem);
-                } else {
-                    $aResults[$sItem]['result'] = false;
-                    $aResults[$sItem]['error'] = gT('User is already a member of the group.');
+            /* check if have permission */
+            if (
+                Permission::model()->hasGlobalPermission('usergroups', 'update') /* Global update permission @see UserGroupController->actionEdit */
+                || $oUserGroup->requestEditGroup($oUserGroup->ugid, Yii::app()->session['loginID'])  /* This user group permission */
```
Comment (gabrieljenik) [inline application/controllers/UserManagementController.php:1120]: Can we use getCurrentUserId()?

Comment (Shnoulle) [reply application/controllers/UserManagementController.php:1120]: Now, yes, not before.

**Pull request merged**